Implement tags search functionality

petark7 opened the pull request:
This PR implements optimized tags search functionality.

edichoska commented:
<img width="1096" alt="Screenshot 2024-08-09 at 00 39 38" src="https://github.com/user-attachments/assets/fb15ee73-8bfb-40c5-98b6-33d049ea5ea4">
<img width="1048" alt="Screenshot 2024-08-09 at 00 40 14" src="https://github.com/user-attachments/assets/85082081-8a85-4c38-9cb9-3439a1781bbf">
If I add space after the word the search is not working.
The columns in the table shouldnt change their height based on how many entries we have.
If on search there is one item, that row takes the whole height of the table. If we have two items, they split the table height in two.
This shouldn't happen.
You should define the height of the rows to prevent this behaviour.

petark7 commented:
>  If I add space after the word the search is not working. 

Why would we want to still show the tag if we add space after? From what I see, this seems kind of unnecessary since I see no use case for it. 

I have fixed the issue with the height of the rows.

petark7 commented:
> > > If I add space after the word the search is not working.
> > 
> > @edichoska   
> > Why would we want to still show the tag if we add space after? From what I see, this seems kind of unnecessary since I see no use case for it.   
> > I have fixed the issue with the height of the rows.
> 
> Thank you for fixing the row issue. It is better from user experience perspective. Since not only developers will be using this platform, but the marketing team will also be using it. We should always try to remove effort/thinking from the user. We should opt for implementing the following: -a fuzzy search or partial matching algorithm that can handle slight variations or typos in the input
> 
> *   trim leading or trailing spaces or ".,  
>     or we can opt for giving feedback to the user. If the user misspells something we can give suggestions: similar to google when you misspell something while you search

Fixed!
